### PR TITLE
Document the LEB128 patch

### PR DESCRIPTION
**Pull request by snicolet**
Add some comments and harmonize style for the LEB128 patch.

closes https://github.com/official-stockfish/Stockfish/pull/4642

No functional change

### CODE REVIEW

File: src/nnue/nnue_common.h [hunk 1]
```diff
@@ -162,77 +166,118 @@ namespace Stockfish::Eval::NNUE {
...
 
+
+  // read_leb_128(s, out, N) : read N signed integers from the stream s, putting them in
+  // the array out. The stream is assumed to be compressed using the signed LEB128 format.
+  // See https://en.wikipedia.org/wiki/LEB128 for a description of the compression scheme.
   template <typename IntType>
   inline void read_leb_128(std::istream& stream, IntType* out, std::size_t count) {
-      static_assert(std::is_signed_v<IntType>, "Not implemented for unsigned types");
+
+      // check that presence of our LEB128 magic string
```
Comment (Sopel97) [inline src/nnue/nnue_common.h:176]: If we're doing any work around here, then I'd suggest we move this check to a separate function that calls read_leb_128. The rationale is that our magic string is a property of tensor serialization, not of the LEB128 compressed data. I envision a `read_tensor` function that checks the magic string and calls `read_leb_128`. Possibly not in scope of this PR.

File: src/nnue/nnue_common.h [hunk 2]
```diff
@@ -162,77 +166,118 @@ namespace Stockfish::Eval::NNUE {
...
+
               std::uint8_t byte = buf[buf_pos++];
               --bytes_left;
               result |= (byte & 0x7f) << shift;
               shift += 7;
-              if ((byte & 0x80) == 0) {
+
+              if ((byte & 0x80) == 0)
+              {
                   out[i] = sizeof(IntType) * 8 <= shift || (byte & 0x40) == 0 ? result : result | ~((1 << shift) - 1);
```
Comment (Sopel97) [inline src/nnue/nnue_common.h:208]: format the ternary? could use parentheses too

File: src/nnue/nnue_common.h (hunk 1 above)
Comment (snicolet) [reply src/nnue/nnue_common.h:176]: yes, probably better to do it in another PR, if any

**Comment by Sopel97** [comment]
looks good to me

**Pull request closed without merging**